.github/workflows/pre_commit_autoupdate.yml

```diff
@@ -58,6 +58,10 @@ jobs:
         run: |
           git checkout main
```

Comment:
Could we change this to only checkout the pre-commit config from main rather than the whole branch? This way we won't need the new step, but the desired logic will still be achieved.

Reply:
Good idea! I've tweaked your suggestion slightly to ensure it's clear what's going on.